projects/bootloader/scripts/test_flash_application_code.py

```diff
@@ -0,0 +1,136 @@
+# pylint: skip-file
+"""This Module Tests methods in flash_application_code.py"""
+
+# Import protobuf
+from can_datagram import DatagramListener
+from can_datagram import DatagramSender
+from can_datagram import Datagram
+import time
+import asyncio
+import can
+import unittest
+from protogen import flash_application_code_pb2 as pb2
+import os
+import sys
+currentdir = os.path.dirname(os.path.realpath(__file__))
+parentdir = os.path.dirname(currentdir)
+sys.path.append(parentdir)
+
+
+TEST_PROTOCOL_VERSION = 1
+TEST_DATAGRAM_TYPE_ID = 1
+TEST_NODES = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+TEST_DATA = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3, 2, 3, 8, 4, 6, 2, 6, 4, 3, 3]
+TEST_CHANNEL = "vcan0"
+
+
+class TestFlashApplication(unittest.TestCase):
```

Comment:
The purpose of a unit test is to test the actual production code, meaning your tests have to use the `FlashApplicationCode` class from the flash_application_code.py. Otherwise, it's not actually verifying that what your code does is correct - you could change the production code however you want and the tests would still pass!

You'll have to find a way to mock the responses you need with the existing code (you might have to modify it to make it more testable). One way to do this would be to have `receive_own_messages` set to True in the test and have your test function instantiate another DatagramListener and DatagramSender which provides the mock messages.